controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -170,17 +170,22 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 				reterr = nil
 			}
 		}
-		// Always attempt to update status.
-		if err := r.updateStatus(ctx, kcp, cluster); err != nil {
-			logger.Error(err, "Failed to update KubeadmControlPlane Status")
-			reterr = kerrors.NewAggregate([]error{reterr, err})
+
+		// If we don't have a DeletionTimestamp set, we try to update status
+		if kcp.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
I'd move this check inside `updateStatus` on 325-ish, right before `GetWorkloadCluster` call